syntax = "proto3";

package dfs;

service DFS {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
    rpc ListDirectories(ListRequest) returns (ListResponse);
    rpc MakeDirectory(MakeDirRequest) returns (MakeDirResponse);
    rpc ChangeDirectory(ChangeDirRequest) returns (ChangeDirResponse);
    rpc RemoveDirectory(RemoveDirRequest) returns (RemoveDirResponse);
}

message ChangeDirRequest {
    string username = 1;
    string target_directory = 2;
}

message ChangeDirResponse {
    bool success = 1;
    string new_directory = 2;
    string message = 3;
}

message RemoveDirRequest {
    string username = 1;
    string directory_name = 2;
}

message RemoveDirResponse {
    bool success = 1;
    string message = 2;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    bool success = 1;
    string message = 2;
}

message CreateUserRequest {
    string username = 1;
    string password = 2;
}

message CreateUserResponse {
    bool success = 1;
    string message = 2;
}

message ListRequest {
    string username = 1;
    string current_directory = 2; 
}

message ListResponse {
    repeated string directories = 1;
}

message MakeDirRequest {
    string username = 1;
    string directory_name = 2;
}

message MakeDirResponse {
    bool success = 1;
    string message = 2;
}

